gsl_simulation/views.py

```diff
@@ -284,46 +293,75 @@ def wrapper(*args, **kwargs):
             logging.error("An error occurred: %s", str(e))
             return JsonResponse(
                 {
-                    "success": False,
                     "error": f"An internal error has occurred : {str(e)}",
-                }
+                },
+                status=400,
             )
 
     return wrapper
 
 
-# TODO pour les fonctions ci-dessous : vérifier que l'utilisateur a les droits nécessaires
+def can_the_user_edit_simulation_projet(func):
+    def wrapper(*args, **kwargs):
+        user = args[0].user
+        if user.is_staff:
+            return func(*args, **kwargs)
+
+        simulation_projet = get_object_or_404(SimulationProjet, id=kwargs["pk"])
+        if not SimulationProjetService.is_simulation_projet_in_perimetre(
+            simulation_projet, user.perimetre
+        ):
+            return JsonResponse(
+                {
+                    "error": "You are not allowed to edit this project",
+                },
+                status=404,
+            )
```

Comment:
Pour éviter un oracle, je conseille d'adopter exactement le même comportement que si la SimulationProjet n'existait pas :

```suggestion
                    raise Http404(
            "No %s matches the given query." % SimulationProjet._meta.object_name
        )
```

(Je suis allée lire la source du cœur de Django)

Dans ce cas je me dis qu'on n'a pas besoin d'une JsonResponse propre : l'utilisateur veut jouer à l'apprenti hacker, c'est pas grave si le JS plantouille sur le retour de la requête.